smurfs/src/reducers/index.js

```diff
@@ -1,7 +1,50 @@
+import { 
+  FETCH_SMURFS, 
+  SUCCESS, 
+  FAILURE, 
+  POST_SMURF, 
+  DELETE_SMURF, 
+  PUT_SMURF,
+} from '../actions';
+
 /*
   Be sure to import in all of the action types from `../actions`
 */
 
+const initialState = {
+  smurfs: [],
+  fetchingSmurfs: false,
+  error: null,
+  editing: false,
+}
+
+export const Reducer = (state = initialState, action) => {
+  switch(action.type){
+    case FETCH_SMURFS:
+      return {
+        ...state,
+        fetchingSmurfs: true,
```

Comment:
I'd add the error property here as well and set it to null during the fetching stage, you can have the case that you are loading and coming back from an error and then you'd need to clean the error value.